src/pybind/mgr/dashboard/module.py

```diff
@@ -751,20 +752,110 @@ def get_perf_schema(self, **args):
                         ret[k1][k2] = sorted_dict
                 return ret
 
-            #Adding method to consolidate the data of all osd into one JSON format.
+            #Method to consolidate the data of all osd into one JSON format.
             @cherrypy.expose
             @cherrypy.tools.json_out()
             def get_counter_allosd(self, path):
                 osd_map = global_instance().get("osd_map")
-                logger.debug(osd_map)
-                result = {}
+                result = []
                 for osd in osd_map['osds']:
                     osd_id = "{0}".format(osd['osd'])
                     counter_data = global_instance().get_counter('osd', osd_id, path)
-                    result[osd['osd']] = counter_data[path]
+                    result.append(counter_data[path])
+
+                return result
+
+            #Method to find the average of consolidated data
+            @cherrypy.expose
+            @cherrypy.tools.json_out()
+            def get_counter_data(self, path):
+                data = self.get_counter_allosd(path)
+                result = []
+                osd_count = len(data)
+
+                for i in range(0,len(data[0])):
+
+                      epoch = data[0][i][0]
+                      values = []
+
+                      for osd_id in range(0,osd_count):
+                            values.append(data[osd_id][i][1])
+
+                      result.append(
+                            [epoch, values])
 
                 return result
 
+            @cherrypy.expose
```

Comment:
why would you want to expose this method?

Reply:
This function is no longer there in the final commit. Again, I would like your opinion on squashing all the changes related to graphs and TS in a single commit.